config/client_config.go

```diff
@@ -80,15 +82,24 @@ func NewClientConfig() *ClientConfig {
 
 // Builds a list of fronted.Providers to use based on the configuration
 func (c *ClientConfig) FrontedProviders() map[string]*fronted.Provider {
+	sniRegion := geolookup.GetCountry(0)
+	if sniRegion == "" {
+		sniRegion = "default"
+	}
 
 	providers := make(map[string]*fronted.Provider)
 	for pid, p := range c.Fronted.Providers {
+		sniConfig := p.FrontingSNIs[sniRegion]
+		if sniConfig != nil && sniConfig.UseArbitrarySNIs && len(sniConfig.ArbitrarySNIs) == 0 {
```

Comment:
We want to fall back to the default SNI config if `sniRegion` does not appear in `p.FrontingSNIs`.  Currently, if there is no entry in `p.FrontingSNIs` for `sniRegion`, then `sniConfig` will never be populated (`sniConfig.UseArbitrarySNIs` will be ignored because `sniConfig != nil` will fail).

I think we instead want something like:
```
sniConfig, ok := p.FrontingSNIs[sniRegion]
if !ok {
  sniConfig = p.FrontingSNIs["default"]
}
```

Or am I misunderstanding something?

Reply:
aaaa I have misunderstood, I thought we only want to use the default arbitrary list, will fix that soon

Reply:
Wait I didn't misunderstood, the provided example does mention that, if there are no arbitrary sni provideda and we want to use it, we want to use it from the default section:

```
fronting-snis:
  default:
    use-arbitrary-snis: false
    arbitrary-snis:
      - example.com
      - example2.com
      - ... etc ...
  ir:
    use-arbitrary-snis: true
    arbirtary-snis:
      - iran-example.com
      - iran-example2.com
      - ... etc ...
  cn:
    use-abritrary-snis: true
    # no abritrary-snis list; use list from default section
```

Reply:
The example looks good and your updated code implements it correctly!